annotation-validator-maven-plugin/src/test/java/uk/gov/justice/plugin/annotations/AnnotationCheckMojoTest.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.justice.plugin.annotations;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.junit.After;
+import uk.gov.justice.domain.annotation.Event;
+import uk.gov.justice.raml.maven.test.utils.BetterAbstractMojoTestCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static com.google.common.base.Joiner.on;
+import static org.apache.commons.io.FileUtils.readLines;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * The tests below rely on a dummy java project with compiled source code which can be found in the following location
+ * /src/test/resources/annotation-plugin
+ * <p>
+ * The project has been compiled and the target directory with compiled classes preserved for running these tests
+ */
+public class AnnotationCheckMojoTest extends BetterAbstractMojoTestCase {
+
+    private static final String PLUGIN_VALIDATION_RESULT_PREFIX = "annotation-plugin-validation-result-";
+    private static final String PROJECT_BASE_DIR = "src/test/resources/annotation-plugin";
+    private static final String REPORT_FILE_EXTENSION = ".csv";
+
+    @After
+    public void tearDown() {
+        for (File f : new File(PROJECT_BASE_DIR).listFiles()) {
+            if (f.getName().startsWith(PLUGIN_VALIDATION_RESULT_PREFIX)) {
+                f.delete();
+            }
+        }
+    }
+
+    public void testShouldScanClassesWithAnnotationsAndGenerateReportWithoutFailingBuild() throws Exception {
```

Comment:
normally we start the test method names with "should"

Reply:
need the test prefix as these MoJO tests are not using the @Test annotation